api/tests/Api/Periods/CreatePeriodTest.php

```diff
@@ -155,6 +155,21 @@ public function testCreatePeriodValidatesInvalidStart() {
         ]);
     }
 
+    public function testCreatePeriodValidatesInvalidStartDateTime() {
+        $this->markTestSkipped(
+            'FIXME: DateTime format is currenctly accepted by mistake.'
```

Comment:
Bisher hatte ich das eher als "Feature" angeschaut, da ein API client sich dann etwas weniger Mühe geben muss, das genau richtige Format auszuliefern. Aber es wäre wohl wirklich verwirrend wenn ein API client bemerkt dass er für den Periodenstart eine Zeit mitsenden kann. Ich weiss nur nicht ob wir das so einfach hinbiegen können, oder ob man dafür tief in API platform oder Symfony Sachen ändern müsste.

Reply:
Ah, ich hatte schon mal diesen [PR](https://github.com/symfony/symfony/pull/42502) gefunden der dazu zuerst nötig ist, und der kommt glaube ich mit Symfony 5.4 oder 6.0. Danach ist noch die Frage wie (oder wann) man das in API platform verwenden kann.

Reply:
In https://github.com/ecamp/ecamp3/pull/2373 funktioniert der Test nun - wenn wir das denn so haben wollen

Reply:
Relatedly, der `ExpressionDateFilter` aus #2062 (und auch der stock API platform DateFilter) unterstützt sowohl ISO-Datum als auch DateTime-Beschreibungen, und sogar auch Werte wie "today" oder "+1 day", i.e. alles was die PHP-Funktion [`strtotime`](https://www.php.net/manual/en/function.strtotime.php) unterstützt. Beim ExpressionDateFilter ist das sogar nützlich und je nach Use Case wichtig: Wenn man die ScheduleEntries nach Tag filtern will, aber den Tages-Cutoff nicht um 0:00 sondern z.B. um 05:00 machen will (wie eCamp2 es macht), dann muss man nicht nur nach einem Datum sondern nach einer genauen Zeit filtern können.
Der Konsistenz halber müssten wir also vielleicht im gleichen Zug den ExpressionDateFilter in ExpressionDateTimeFilter umbenennen? Den DateFilter von API Platform können wir nicht umbenennen, aber wir könnten eine Subklasse davon erstellen die nichts ändert ausser den Namen.

Reply:
Wäre wahrscheinlich konsistenter. Soll ich das gleich in einen der PR #2373 mit aufnehmen?

Den Stock `DateFilter` verwenden wir aktuell nicht, oder? Von dem her könnten wir uns dort das Renaming noch sparen. Oder brauch wir den sowieso bald?

Reply:
Ja, klingt nach einem Plan. `DateFilter` wüsste ich momentan auch keinen Ort wo wir den verwenden. Könnte allenfalls in Zukunft mal auf Period erlaubt werden, aber für unser Frontend wohl nicht MVP-relevant.